### DISCUSSION TITLE
MvvmHelpers.AutowireViewModel() protection level

### DISCUSSION THREAD

**Discussion by Chorusapp** [PrismLibrary/Prism#2504] [WPF]
I am writing a ScopedRegionNavigationContentLoader class like Brian did in his course on Pluralsight, Prism Problems & Solutions: Mastering TabControl.

I copied the code from RegionNavigationContentLoader class and pasted it up to my class, ScopedRegionNavigationContentLoader.

I modified some code like Brian did.

But, MvvmHelpers.AutowireViewModel() protection level is internal so that I can't access it on CreateNewRegionItem() of my class.

What's the solution?

![1](https://user-images.githubusercontent.com/30747047/124292193-45567680-db90-11eb-8e4d-d5ff1e4c3df2.png)

**Comment by brianlagunas** [comment]
Use the actual attached property instead. but I doubt you have to copy the entire class since those methods are now virtual and can be overridden in a derived class now.

**Reply by Chorusapp** [reply]
I didn't realize that.

Thanks a lot.
